usr/src/cmd/zdb/zdb.c

```diff
@@ -3164,8 +3164,8 @@ zdb_dump_block(char *label, void *buf, uint64_t size, int flags)
 	for (i = 0; i < nwords; i += 2) {
 		(void) printf("%06llx:  %016llx  %016llx  ",
 		    (u_longlong_t)(i * sizeof (uint64_t)),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i]) : d[i]),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i + 1]) : d[i + 1]));
+		    (u_longlong_t)BE_64(d[i]),
+		    (u_longlong_t)BE_64(d[i + 1]));
```

Comment:
The current code only works if the block consists entirely of 64-bit numbers.  For any other kind of block, it prints nonsense.  Given that most LE engineers are proficient at mentally byteswapping hex dumps, I think zdb should print something meaningful in every case rather than optimizing for the 64-bit number case and failing for every other case.  This is also the default behavior of most hexdump utilities: dump bytewise.  Given that the original code was written in early 2006, it may well have been written on a SPARC with the author unaware of the issues on little endian filesystems.